format/wav/Reader.hx

```diff
@@ -100,15 +100,38 @@ class Reader {
 		// data
 		if (nextChunk != "data")
 			throw "expected data subchunk";
-		
+
 		var datalen = readInt();
-		
-		// Some files report an incorrect length, so we'll
-		// read the whole file, then subtract if necessary
-		var data = i.readAll ();
-		if (data.length > datalen) 
-			data = data.sub (0, datalen);
-		
+
+		var data = i.read(datalen);

```

Comment:
This might break because of my comment in previous version.
Maybe we should allocate a datalen bytes, read into it once using readBytes then substract if we couldn't read all.